examples/cpp/beta/local/filter-data.cpp

```diff
@@ -108,4 +108,76 @@ TEST_CASE("set up tests", "[write]") {
     CHECK(projects.size() == 0);
     CHECK(items.size() == 0);
 }
+
+TEST_CASE("sort list and results", "[write]") {
+    auto relative_realm_path_directory = "beta_project_items/";
+    std::filesystem::create_directories(relative_realm_path_directory);
+    std::filesystem::path path = std::filesystem::current_path().append(relative_realm_path_directory);
+    path = path.append("project_and_item_objects");
+    path = path.replace_extension("realm");
+
+    auto config = realm::db_config();
+    config.set_path(path);
+    auto realm = db(std::move(config));
+
+    auto item1 = Beta_Item {
+        .name = "Save the cheerleader",
+        .assignee = std::string("Peter"),
+        .isComplete = false,
+        .priority = 6,
+        .progressMinutes = 30
+    };
+    
+    auto item2 = Beta_Item {
+        .name = "Save the world",
+        .assignee = std::string("Peter"),
+        .isComplete = false,
+        .priority = 7,
+        .progressMinutes = 90
+    };
+
+    auto project = Beta_Project {
+        .name = "Heroes Project"
+    };
+
+    project.items.push_back(&item1);
+    project.items.push_back(&item2);
+
+    realm.write([&] {
+        realm.add(std::move(project));
+    });
+
+    // :snippet-start: sort-results-by-single-property
+    auto items = realm.objects<Beta_Item>();
+    CHECK(items.size() == 2); // :remove:
+
+    // Sort with `false` returns objects in descending order.
+    auto itemsSorted = items.sort("priority", false);
+    // :snippet-end:
+    // The item titled "Save the world" should be the first item since we are sorting in
+    // descending order based on priority.
+    CHECK(itemsSorted[0].name == "Save the world");
+
+    auto projects = realm.objects<Beta_Project>();
+    CHECK(projects.size() == 1); // :remove:
+    auto specificProject = projects[0];
+    
+    // :snippet-start: sort-list-by-multiple-properties
+    auto sortedListProperty = specificProject.items.sort({{"assignee", true}, {"priority", false}});
```

Comment:
possible to get this on multiple lines so it's easier to see the multiple sorts? something like:
```suggestion
auto sortedListProperty = specificProject.items.sort(
    {{"assignee", true},
     {"priority", false}}
);
```